packages/block-editor/src/components/use-moving-animation/index.js

```diff
@@ -1,35 +1,32 @@
 /**
  * External dependencies
  */
-import { useSpring } from '@react-spring/web';
+import { Controller } from '@react-spring/web';
 
 /**
  * WordPress dependencies
  */
-import {
-	useState,
-	useLayoutEffect,
-	useReducer,
-	useMemo,
-	useRef,
-} from '@wordpress/element';
-import { useReducedMotion } from '@wordpress/compose';
+import { useLayoutEffect, useMemo, useRef } from '@wordpress/element';
 import { getScrollContainer } from '@wordpress/dom';
+import { useSelect } from '@wordpress/data';
 
 /**
- * Simple reducer used to increment a counter.
- *
- * @param {number} state Previous counter value.
- * @return {number} New state value.
+ * Internal dependencies
  */
-const counterReducer = ( state ) => state + 1;
+import { store as blockEditorStore } from '../../store';
 
-const getAbsolutePosition = ( element ) => {
+/**
+ * If the block count exceeds the threshold, we disable the reordering animation
+ * to avoid laginess.
+ */
+const BLOCK_ANIMATION_THRESHOLD = 200;
```

Comment:
By moving this into `useMovingAnimation`, the moving animation is now prevented in the list view when the global block count is greater than 200. Was that intentional?

I think for the list view case, where list items are displayed conditionally based on windowing logic, we might not need to guard it behind this global block count threshold. The reason I'm thinking about this is because over in https://github.com/WordPress/gutenberg/pull/56625 I'm exploring displacing list view items when dragging within the list view, and `useMovingAnimation` provides a handy way of smoothly animating a dropped list view item into its final position.

I noticed while editing the blog home template in TT4 that the `getGlobalBlockCount()` already starts out pretty high (165 in my test site) so it's quite easy to get to over 200 and then lose the animation.